pkg/yang/entry.go

```diff
@@ -1477,10 +1478,10 @@ func (s sortedErrors) Less(i, j int) bool {
 		case len(fj) < x && len(fi) < x:
```

Comment:
This looks like a woefully under-documented part of the code, thanks for looking for a fix!

I think this line should be `case len(fj) == x && len(fi) == x:` to indicate that neither error slice has this substring index after the split. It's the same behaviour but more readable to me.

Reply:
I don't think so. If this, it will crash because of out of index.

Reply:
Good point -- the algorithm will try again (since it returns 0) and error out next time. I thought it would magically exit the for loop when it sees this condition.

Maybe you can add a comment here `// If neither element exist, then they're equal`.

Reply:
Okey.  I've reread the code, in this case, I think it's better to return  `1` replaced ` 0` because it will exit the for loop immediately without going into the next loop.

Reply:
That makes sense. However, because doing this is somewhat unintuitive for the reader, because "1" here means >= instead of ==. So I would add to the comment above `compare := func(x int) int {` to say
`// -1 means less than, 1 means greater or *equal* to, and 0 means equal thus far, but still undecided and need to see the rest of the elements.